index.src.html

```diff
@@ -705,73 +759,75 @@ <h4 id="fetching">Fetching</h4>
   1.  Define a new <dfn export>Private Network Access check</dfn> algorithm.
       Given a [=request=] |request| and a [=connection=] |connection|:
 
-      1. If |request|'s [=request/origin=] is a [=potentially trustworthy
-         origin=] and |request|’s [=request/current URL=]’s [=request/origin=]
-         is [=same origin=] with |request|’s [=request/origin=], then return
-         null.
-
-      1.  If |request|'s [=request/client=] is not a [=secure context=]
-          (including if it is null):
-
-          1.  Let |clientAddressSpace| be |request|'s
-              [=request/policy container=]'s
-              [=policy container/IP address space=].
+      1.  If |request|'s [=request/origin=] is a [=potentially trustworthy
+          origin=] and |request|’s [=request/current URL=]’s [=request/origin=]
+          is [=same origin=] with |request|’s [=request/origin=], then return
+          null.
 
-          1.  If |connection|'s [=connection/IP address space=] is
-              [=IP address space/less public=] than |clientAddressSpace|, then
-              return a [=network error=].
+      1.  If |request|'s [=request/policy container=] is null, then return null.
 
-          1.  Return null.
+          NOTE: If |request|'s [=request/policy container=] is null, then PNA
+          checks do not apply to |request|. Users of the [=fetch=] algorithm
+          should take care to either set |request|'s [=request/client=] to an
+          [=environment settings object=] with a non-null [=environment settings
+          object/policy container=] and let [=fetch=] initialize |request|'s
+          [=request/policy container=] accordingly, or to directly set
+          |request|'s [=request/policy container=] to a non-null value.
 
-          NOTE: If |request|'s [=request/policy container=] is null, then PNA checks
-          do not apply to |request|. Users of the [=fetch=] algorithm should take
-          care to either set |request|'s [=request/client=] to an
-          [=environment settings object=] with a non-null
-          [=environment settings object/policy container=] and let [=fetch=]
-          initialize |request|'s [=request/policy container=] accordingly, or to
-          directly set |request|'s [=request/policy container=] to a non-null value.
+      1.  Let |clientAddressSpace| be |request|'s [=request/policy container=]'s
+          [=policy container/IP address space=]
 
-      1.  Let |clientAddressSpace| be |request|'s
-          [=request/policy container=]'s
-          [=policy container/IP address space=].
+      1.  If |request|'s [=request/client=] is not a [=secure context=]
+          (including if it is null), and |connection|'s [=connection/IP address
+          space=] is [=IP address space/less public=] than |clientAddressSpace|,
+          then return a [=network error=].
 
       1.  If |request|'s [=request/target IP address space=] is not null, then:
```

Comment:
I think that in Chromium's implementation, we check target IP address space before comparing against the client IP address space. We can do that because a request's target IP address space is only set when the client is a secure context. We can and probably should check that when initializing a fetch request from `RequestInit`, since the new option is useless for non-secure contexts?

If we move this to after the target IP address space check, then we can combine it with the other "less-public" branch. I believe we will only want to populate the network error's IP address space for secure contexts, because other parts of this spec depend on that to trigger a preflight - and we do not wish to trigger preflights for non-secure contexts.